Hotfix 25.0.0.1

BobbyRBruce opened the pull request:
This adds the following to the `stable` branch as a hotfix release (v25.0.0.1)


- #2492 
- #2422 
- #2399 
- #2397 
- #2415 
- #2441 
- #2464
- #2502 
- #2512

erin-le committed:
util: bump urllib3 to 2.5.0 in util/gem5-resources-manager (#2397)

erin-le committed:
stdlib: remove duplicate ClassicGeneratorExitHandler (#2415)

Previously, there were two definitions of the
ClassicGeneratorExitHandler, which handles hypercall 0, in
`exit_handler.py`. This commit removes the definition located earlier in
the file, as it isn't used.

TomXia committed:
arch-vega: Fix incorrect address translation caused by tlb

This commit fixes an issue where the tlb returns incorrect
physical address if the pagesize is other than 4KB. Now,
implicit copy constructor is used to create the new
VegaTlbEntry from the post-translated results, instead of
the assumed 4k-aligned address.

giactra committed:
arch-arm: Add FEAT_FP16 FP instructions (#2441)

With PR 2071 [1] we introduced FEAT_FP16 into gem5. While it covered
most SIMD FP16 instructions, it left out most of the scalar ones.

This commit complements the original PR by providing the AArch64 scalar
floating point implementation of FEAT_FP16

[1]: https://github.com/gem5/gem5/pull/2071

Change-Id: I60837911eedebc2b44590b17c163dee7b6819df3

Signed-off-by: Giacomo Travaglini <giacomo.travaglini.arm.com>
Co-authored-by: Giacomo Travaglini <giacomo.travaglini.arm.com>

studyztp committed:
cpu: Fix looppoint analysis v25 (#2399)

This is a PR for the issue: https://github.com/gem5/gem5/issues/2395

The new probe structure uses ```dynamic_cast``` to check
the```ProbeListener```, which caused a casting mismatch with the
LoopPoint Analysis ProbeListener, because the LoopPoint Analysis
ProbeListener uses ```StaticInstPtr``` instead of ```const
StaticInstPtr``` as part of its Arg while the Probe Point that it is
listening to, ```Commit```, uses ```const StaticInstPtr```.
This commit fixes the issue.

saul44203 committed:
arch-riscv: populate logBytes/paddr after functional pt walk (#2464)

Both `logBytes` and `addr` references were not updated after finishing a
functional translation walk in the RISC-V page table walker.

Should fix #2410.

arichardson committed:
arch-arm: fix writeback type for AArch32 FP16 instructions (#2492)

Fixes warning: implicit conversion from 'uint32_t' (aka 'unsigned int')
to 'float' may lose precision [-Wimplicit-int-float-conversion]

Change-Id: I605a2db2bf46e80b826a8c9e672e018109de478c

BobbyRBruce committed:
base,misc: Up version to 25.0.0.1

BobbyRBruce committed:
misc: Add v25.0.0.1 hotfix release notes

giactra commented:
Thanks Bobby, could you include this PR as well? https://github.com/gem5/gem5/pull/2441

powerjg commented:
Also #2464

BobbyRBruce commented:
#2564 and #2441 have been added.

erin-le commented:
https://github.com/gem5/gem5/pull/2474 also needs to be added once it passes all of the CI tests and is merged into `develop`.

BobbyRBruce commented:
> #2474 also needs to be added once it passes all of the CI tests and is merged into `develop`.

Good catch! I'll wait for it to be merged then cherry pick it.

dhschall commented:
#2502 fixes the same as #2399. Might be worth to merge this as well

dhschall committed:
cpu: Adapt simpoint listener to new probe structure (#2502)

The new probe structure expects ```const StaticInstPtr``` when
listening to the ```Commit``` probe point. Simpoints still used
the ```StaticInstPtr``` which causes a casting mismatch.

The same issue was raised for looppoint analysis in #2399

powerjg commented:
Got another one!

https://github.com/gem5/gem5/pull/2512

nmosier committed:
cpu-o3: properly index time buffer when clearing states (#2512)

Fix #2402.

Change-Id: I3b61919547fae21a9ed69d4a46e5a3903e41505c
